### PR TITLE
TT-2977 bugfix/validate names claims saml

### PR DESCRIPTION
**Pull request by sredxny**
## Description

Validate claims, not just assume that we receive the data, but add some conditions when the data is not sent by the idp or bad mapped by the user in thr TIB profile. If names are not received, then the email will be used as name

## Related Issue
https://tyktech.atlassian.net/browse/TT-2977

## Motivation and Context
Respond correctly when we miss some data from IDP and do not crash application

## How This Has Been Tested
- Ran tests
- Run SAML Flow using Azure: 2 use-cases, receiving and not receiving the surname and givenname in the claims.

Profile Used:
```
[{
    "ID": "saml-sso-dash-login",
    "OrgID": "5e2091c4d4aefce60c04fb92",
    "ActionType": "GenerateOrLoginUserProfile",
    "Type": "redirect",
    "ProviderName": "SAMLProvider",
    "ProviderConfig" : {
        "CertLocation": "./full-cert.pem",
        "SAMLBaseURL": "https://ba68-186-29-99-80.ngrok.io/",
        "ForceAuthentication": false,
        "FailureRedirect": "https://ba68-186-29-99-80.ngrok.io/?fail=true",
        "IDPMetaDataURL": "https://login.microsoftonline.com/my-id"
    },
    "IdentityHandlerConfig" : {
        "DashboardCredential" : "809dba3548cd49a26087ed0152703839"
    },
    "ReturnURL" : "http://tyk-dashboard:3000/tap"
}]
```
With claims:
- Go to your azure app and start the flow by clicking 'Test' button
- Input your credentials
- You should be redirected to dashboard and your name+lastname are shown in the upper right corner.

Without Claims:
- Go to azure and deactivate the givenname and surname from the claims, and save changes
- Start the flow by clicking the 'Test' button
- Input your credentials
- You should be redirected to the dashboard, but this time your email is displayed in the upper right corner because the names were not received from the IDP. The application should not panic

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [ ] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
  - [ ] If new config option added, ensure that it can be set via ENV variable
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] When updating library version must provide reason/explanation for this update.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Check your code additions will not fail linting checks:
  - [ ] `go fmt -s`
  - [ ] `go vet`

### CODE REVIEW

File: providers/saml.go [hunk 1]
```diff
@@ -371,3 +363,40 @@ func PrintErrorStruct(err error) {
...
+	// so lets make it configurable with a sensible backup
+	if emailClaim == "" {
+		emailClaim = DefaultEmailClaim
+	}
+
+	if _, ok := claims[emailClaim]; ok {
+		email = claims[emailClaim].(string)
+	} else if _, ok := claims["http://schemas.xmlsoap.org/ws/2005/05/identity/claims/"]; ok {
+		// fallback to WIF
+		email = claims[WIFUniqueName].(string)
```
Comment (tbuchaillot) [inline providers/saml.go:398]: I think it would be good if we double check that claims map have `WIFUniqueName ` value before casting

**Review by tbuchaillot — CHANGES_REQUESTED** [review]
@sredxny left a small comment on a side scenario, but overall it looks 👌

File: providers/saml.go (hunk 1 above)
Comment (sredxny) [reply providers/saml.go:398]: @tbuchaillot makes 100% sense

Comment (sredxny) [reply providers/saml.go:398]: @tbuchaillot done!

**Review by tbuchaillot — APPROVED** [review]

**Pull request merged**